insights/parsers/package_provides_java.py

```diff
@@ -0,0 +1,49 @@
+"""
+PackageProvidesJava - Command
+=============================
+
+This parser reads the output from pre_command
+``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; \
+do echo $jp `readlink -e $jp | xargs rpm -qf`; done``
+and coverts it into a dict that links running_java_command with its package name.
+The reason why using above pre_command is that we need to record the links between running_java_command
+and package which provides the java command. In ``ps aux`` output, we can only get what java command starts a
+java application, instead of java package.
+Through this way, when there is jdk bug, we can detect whether a running java application will be affected.
+
+
+Typical contents of the pre_command::
+
+    /usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64
+
+Parsed result::
+
+    {"command": "/usr/lib/jvm/jre/bin/java", "package": "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"}
+
+Examples:
+
+    >>> command_package = shared[PackageProvidesJava]
+    >>> command_package.data
+    {"command": "/usr/lib/jvm/jre/bin/java", "package": "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"}
+    >>> command_package.get("command")
+    '/usr/lib/jvm/jre/bin/java'
+"""
+
+from insights import parser, Parser, LegacyItemAccess
+
+
+@parser('package_provides_java')
+class PackageProvidesJava(Parser, LegacyItemAccess):
+    """
+    Parse the output of pre_command
+    ``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; \
+    do echo $jp `readlink -e $jp | xargs rpm -qf`; done``.
+    """
+
+    def parse_content(self, content):
+        comamnd_package_dict = {}
+        for line in content:
+            l = line.split()
+            if len(l) == 2:
+                comamnd_package_dict = ({"command": l[0], "package": l[1]})
+        self.data = comamnd_package_dict
```

Comment:
Better add two `Attributes` directly.

self.command = l[0]
self.package = l[1]

Then you do not need to mixin LegacyItemAccess

Reply:
Have updated according to this comment.